src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/killaura/features/AutoBlock.kt

```diff
@@ -80,9 +83,36 @@ object AutoBlock : ToggleableConfigurable(ModuleKillAura, "AutoBlocking", false)
         blockVisual = true
     }
 
+    fun shouldUnblockToHit(): Boolean {
+        return unblockMode == UnblockMode.NONE
+    }
+
+    fun shouldBlock(): Boolean {
+        return if (blockMode == BlockMode.WATCHDOG) {
+            if (tickOn == 0) {
+                true
+            } else {
+                network.sendPacket(Full(player.x, player.y, player.z, player.yaw, player.pitch, player.isOnGround))
+                network.sendPacket(
+                    PlayerActionC2SPacket(
+                        PlayerActionC2SPacket.Action.STOP_DESTROY_BLOCK,
+                    player.blockPos,
+                    player.horizontalFacing.opposite)
+                )
+                true
+            }
+        } else {
+            tickOn == 0
+        }
+    }
+
     /**
      * Starts blocking.
      */
+    @Suppress("CognitiveComplexMethod")
+    // I kid you not,
+    // "too complex based on Cognitive Complexity (complexity: 15).
+    // Defined complexity threshold for methods is set to '15'"
```

Comment:
Try replacing this part of the code:
```
        val blockHand = if (canBlock(player.mainHandStack)) {
            Hand.MAIN_HAND
        } else if (canBlock(player.offHandStack)) {
            Hand.OFF_HAND
        } else {
            // We cannot block with any item.
            return
        }
```

with this one:

```
        val blockHand = when {
            canBlock(player.mainHandStack) -> Hand.MAIN_HAND
            canBlock(player.offHandStack) -> Hand.OFF_HAND
            else -> return  // We cannot block with any item.
        }
```
It should lower the complexity =)